erpnext/stock/doctype/item/item.py

```diff
@@ -489,6 +489,15 @@ def update_show_in_website(self):
 		if self.disabled:
 			self.show_in_website = False
 
+		variants = frappe.get_all("Item", filters={"variant_of": self.name}, fields=["item_code"])
+
+		if not self.disabled and self.has_variants and not self.show_in_website and variants:
+			frappe.db.sql("""
+				UPDATE `tabItem`
+				SET show_variant_in_website=0
+				WHERE variant_of=%s
+			""", (self.name))
```

Comment:
Avoid using SQL updates, that bypasses any other validations that might need to happen (now or later) and sets it directly in the db. That might cause unexpected behaviour.

```suggestion
		if self.disabled:
			self.show_in_website = False
		elif self.has_variants and not self.show_in_website:
			variants = frappe.get_all("Item", filters={"variant_of": self.name})
			if variants:
				for item in variants:
					item_doc = frappe.get_doc("Item", item.name)
					item_doc.show_variant_in_website = False
					item_doc.save()
```

Reply:
So will have to add this to background jobs if variants are more than 30, it'll take forever to save the doc.

Reply:
That's fair, I think we can safely move this to background jobs